docs/tutorial/fastapi.rst

```diff
@@ -0,0 +1,419 @@
+    for you.
+
+    The size of your rock may vary depending on factors like the architecture
+    you are building on and the packages installed at the time of packing.
+
+Run the FastAPI rock with Docker
+================================
+
+We already have the rock as an `OCI <OCI_image_spec_>`_ archive. Now we
+need to load it into Docker:
+
+.. literalinclude:: code/fastapi/task.yaml
+    :language: bash
+    :start-after: [docs:skopeo-copy]
+    :end-before: [docs:skopeo-copy-end]
+    :dedent: 2
+
+Check that the image was successfully loaded into Docker:
+
+.. literalinclude:: code/fastapi/task.yaml
+    :language: bash
+    :start-after: [docs:docker-images]
+    :end-before: [docs:docker-images-end]
+    :dedent: 2
+
+The output should list the FastAPI container image, along with its tag, ID and
+size:
+
+..  code-block:: text
+    :class: log-snippets
+
+    REPOSITORY            TAG       IMAGE ID       CREATED       SIZE
+    fastapi-hello-world   0.1       30c7e5aed202   2 weeks ago   193MB
+
+.. note::
+    The size of the image reported by Docker is the uncompressed size which is
+    larger than the size of the compressed ``.rock`` file.
+
+Now we're finally ready to run the rock and test your containerised FastAPI
+application:
+
+.. literalinclude:: code/fastapi/task.yaml
+    :language: text
+    :start-after: [docs:docker-run]
+    :end-before: [docs:docker-run-end]
+    :dedent: 2
+
+Use the same ``curl`` command as before to send a request to the FastAPI
+application's root endpoint which is running inside the container:
+
+.. literalinclude:: code/fastapi/task.yaml
+    :language: text
+    :start-after: [docs:curl-fastapi-rock]
+    :end-before: [docs:curl-fastapi-rock-end]
+    :dedent: 2
+
+The FastAPI application should again respond with ``{"message":"Hello World"}``.
+
+View the application logs
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When deploying the FastAPI rock, you can always get the application logs via
+``pebble``:
+
+.. literalinclude:: code/fastapi/task.yaml
+    :language: text
+    :start-after: [docs:get-logs]
+    :end-before: [docs:get-logs-end]
+    :dedent: 2
+
+As a result, :ref:`pebble_explanation_page` will give you the logs for the
+``fastapi`` service running inside the container.
+You should expect to see something similar to this:
+
+..  code-block:: text
+    :class: log-snippets
+
+    2024-10-01T06:32:50.180Z [fastapi] INFO:     Started server process [12]
+    2024-10-01T06:32:50.181Z [fastapi] INFO:     Waiting for application startup.
+    2024-10-01T06:32:50.181Z [fastapi] INFO:     Application startup complete.
+    2024-10-01T06:32:50.182Z [fastapi] INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
+    2024-10-01T06:32:58.214Z [fastapi] INFO:     172.17.0.1:55232 - "GET / HTTP/1.1" 200 OK
+
+You can also choose to follow the logs by using the ``-f`` option with the
+``pebble logs`` command above. To stop following the logs, press ``ctrl+c``.
```

Comment:
Nitpick: This line is ``ctrl+c`` (with a lowercase c) while previous instructions used ``ctrl+C``. Do we want to remain consistent in lower- or uppercase with these instructions? 

P.S. In the Charmcraft Flask UX research, someone said that they believed it's incorrect to use ``ctrl+C`` (specifically with an uppercase C) and suggested to change to lowercase c. I don't know if there's a best practice in this instance, but I thought I would mention it.

Reply:
changed to lower case c

Reply:
@erinecon The practice is to use the form "Key + Key". https://docs.ubuntu.com/styleguide/en/#keyboard-shortcuts-and-keyboard-keys

Reply:
@medubelko Thanks, I guess that doesn't really resolve whether the "C" needs to be upper- or lower-case, but assuming lowercase for now, the line should read ``ctrl + c`` with spaces added

Reply:
Updated the formatting to include spaces